pkg/pool-stable-phantom/contracts/StablePhantomPool.sol

```diff
@@ -1167,6 +1169,8 @@ contract StablePhantomPool is
         uint256[] memory arguments,
         function(uint256, uint256) pure returns (uint256) mutation
     ) private pure {
+        InputHelpers.ensureInputLengthMatch(toMutate.length, arguments.length);
```

Comment:
This just increases runtime cost for no reason. We shouldn't ever be passing in arrays which we don't _know_ are the same length.

Reply:
I agree we *shouldn't*, except we just were :) I put it in to make sure there wasn't another issue. I can take it out, but we should make sure the tests cover this (at least indirectly). The fee amounts should eventually be wrong if we try different token combinations, at least before and after the BPT token.

Reply:
Yeah, to be clear I'm in favour of us having stuff which makes noise if we hit unexpected behaviour, just that we should remove these from the final build before we deploy. We could just make this a TODO to remove once tests are more comprehensive.

Reply:
Well, I already removed it, hoping we could quietly merge and never speak of this again :)

I hesitate to make notes to do things "later," knowing that all too often, later never comes. I'd definitely keep it in your local copy while we're writing the tests, though.